### ISSUE TITLE
Cooldown coordinator breaks vanilla hopper timings

### ISSUE THREAD

**Issue by sarim** [gniftygnome/cooldown-coordinator#2]
Minecraft 1.20.1 Fabric Loader 0.15.11
ModList:
- architectury-9.2.14-fabric.jar
- cloth-config-11.1.118-fabric.jar
- cooldown-coordinator-0.6.2.jar
- fabric-api-0.92.1+1.20.1.jar
- Jade-1.20-fabric-11.9.2.jar
- JadeAddons-1.20.1-fabric-5.2.5.jar
- Powah-5.0.5.jar
- RoughlyEnoughItems-12.1.725-fabric.jar

### Test setup 1: Without cooldown coordinator
hopper send 1 item for both powah orb and chest. Working as intended.

https://github.com/gniftygnome/cooldown-coordinator/assets/1235888/52621c43-e454-40f2-b699-2fa366800baa

### Test setup 2: With cooldown coordinator
hopper send 4 item at once to powah orb and 1 to chest. Obviously broken. I've seen same bug occurs without any redstone lock but when hopper can't send to machine bcz its full, but at soon as its empty, it fills it up in one tick.

https://github.com/gniftygnome/cooldown-coordinator/assets/1235888/5d21d00f-dbf6-42e5-9c4f-b3877f255263

Discord discusstion ref: https://discord.com/channels/1016206797389975612/1016214573667328041/1245063002827264122

**Comment by gniftygnome** [comment]
Cooldown Coordinator is not breaking *vanilla* behavior here.  There is only one thing CC does different from Fabric Transfer API, that one thing is in the non-vanilla code path, and in any case CC's hopper mixin will never move more than one item.  When CC is present, the Transfer API's Hopper insert mixin does not run, so the culprit is not the Transfer API, either.  By elimination, either Architectury API or Powah! itself is performing the surplus moves.

The "why" is the more interesting part.  When the Transfer API's Hopper insert mixin fails to move an item (which implies vanilla cannot move one either), the Transfer API **does not** cancel but instead allows vanilla to attempt the move anyway (and fail).  As a performance optimization, CC **does** cancel in the same situation, preventing both the Transfer API and vanilla from trying moves we already know will fail.  Disabling this optimization resolves the duplicate moves.

Something in a mod (and the only ones I'm testing with are CC, Fabric API, Arch API, Cloth Config, and Powah!) is dependent on a mixin to the Hopper transfer method and when that mixin does not execute, it blindly shovels items, apparently one per tick.  I might trawl through the source code of Arch API and Powah! if I get some time.

BTW, I cannot seem to follow the Discord link you provided.  (I also can't seem to join the AE2 Discord channel for some reason.)  What Discord and channel were you linking, and can you summarize briefly who said what?

**Comment by sarim** [comment]
@gniftygnome  The issue isn't with Arch API or Powah I think. Here is another more smaller example.

Modlist:
cooldown-coordinator-0.6.2.jar
ExtendedDrawers-2.1.1+mc.1.20.1.jar
fabric-api-0.92.1+1.20.1.jar

No redstone involved. But see the speed.

https://github.com/gniftygnome/cooldown-coordinator/assets/1235888/0171627e-2e0b-4e6e-9f58-c075742421a6

**Comment by sarim** [comment]
> BTW, I cannot seem to follow the Discord link you provided. (I also can't seem to join the AE2 Discord channel for some reason.) What Discord and channel were you linking, and can you summarize briefly who said what?

It was in #omnihopper channel in enjarai's library discord. Nothing technically important was said there. 
![image](https://github.com/gniftygnome/cooldown-coordinator/assets/1235888/f5d24641-b188-4b52-abad-7866d271ec47)

**Comment by gniftygnome** [comment]
Cooldown Coordinator 0.6.3 (for Minecraft 1.19 - 1.20.4) and 0.7.2 (for Minecraft 1.20.5 and above) have been released to address this issue.
